src/viur/core/request.py

```diff
@@ -568,73 +570,84 @@ def findAndCall(self, path: str, *args, **kwargs) -> None:
+                caller_kwargs = {}  # The dict of new **kwargs we'll pass to the caller
+                caller_args = []  # List of new *args we'll pass to the caller
 
                 # FIXME: Use inspect.signature() for all this stuff...
-                argsOrder = caller.__code__.co_varnames[:caller.__code__.co_argcount]
+                argsOrder = caller._func.__code__.co_varnames[:caller._func.__code__.co_argcount]
                 # In case of a method, ignore the 'self' parameter
-                if inspect.ismethod(caller):
+                if inspect.ismethod(caller._func):
                     argsOrder = argsOrder[1:]
 
                 # Map args in
                 for idx in range(0, min(len(self.args), len(argsOrder))):
                     paramKey = argsOrder[idx]
                     if paramKey in annotations:  # We have to enforce a type-annotation for this *args parameter
                         _, newTypeValue = self.processTypeHint(annotations[paramKey], self.args[idx], True)
-                        newArgs.append(newTypeValue)
+                        caller_args.append(newTypeValue)
                     else:
-                        newArgs.append(self.args[idx])
-                newArgs.extend(self.args[min(len(self.args), len(argsOrder)):])
+                        caller_args.append(self.args[idx])
+                caller_args.extend(self.args[min(len(self.args), len(argsOrder)):])
                 # Last, we map the kwargs in
                 for k, v in kwargs.items():
                     if k in annotations:
                         newStrValue, newTypeValue = self.processTypeHint(annotations[k], v, False)
                         self.kwargs[k] = newStrValue
-                        newKwargs[k] = newTypeValue
+                        caller_kwargs[k] = newTypeValue
                     else:
-                        newKwargs[k] = v
+                        caller_kwargs[k] = v
             else:
-                newArgs = self.args
-                newKwargs = self.kwargs
-            if (conf["viur.debug.traceExternalCallRouting"] and not self.internalRequest) or conf[
-                "viur.debug.traceInternalCallRouting"]:
-                logging.debug("Calling %s with args=%s and kwargs=%s" % (str(caller), str(newArgs), str(newKwargs)))
-            res = caller(*newArgs, **newKwargs)
-            res = str(res).encode("UTF-8") if not isinstance(res, bytes) else res
+                caller_args = self.args
+                caller_kwargs = self.kwargs
+
+            if (conf["viur.debug.traceExternalCallRouting"] and not self.internalRequest
+                    or conf["viur.debug.traceInternalCallRouting"]):
+                logging.debug(f"Calling {caller=} with {args=}, {kwargs=}")
+
+            res = caller(*caller_args, **caller_kwargs)
+
+            if not isinstance(res, bytes):  # Convert result into str if not bytes!
```

Comment:
`encode()` returns `bytes`.
```suggestion
            if not isinstance(res, bytes):  # Convert the result to bytes if it is not already!
```